.github/workflows/unit-and-integration-tests.yaml

```diff
@@ -0,0 +1,169 @@
+name: Intel Device Plugins K8s Operator Unit and Integration Tests
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  call-inclusive-naming-check:
+    name: Inclusive naming
+    uses: canonical/inclusive-naming/.github/workflows/woke.yaml@main
+    with:
+      fail-on-error: "true"
+  
+  lint-and-unit-tests:
+    name: Lint and unit tests
+    uses: charmed-kubernetes/workflows/.github/workflows/lint-unit.yaml@main
+    with:
+      python: "['3.10', '3.11']"
+    needs:
+      - call-inclusive-naming-check
+
+  integration-tests:
+    name: Integration tests on Testflinger instance with Intel GPU
+    runs-on: [self-hosted, testflinger]
+    steps:
+      - name: Submit job to testflinger
+        uses: canonical/testflinger/.github/actions/submit@main
+        with:
+          poll: true
+          job: |
+            job_queue: dell-precision-3470-c30322
+            provision_data:
+              distro: jammy
+              user_data: |
+                #cloud-config
+                # Note, much of this is borrowed from the
+                # charm-dev multipass template:
+                # https://github.com/canonical/multipass-blueprints/blob/main/v1/charm-dev.yaml
+                package_update: true
+                package_upgrade: true
+                package_reboot_if_required: true
+
+                packages:
+                - python3-pip
+
+                snap:
+                  commands:
+                  # Juju 3.5 is supported until Jan 2025
+                  # https://juju.is/docs/juju/roadmap
+                  - snap install juju --channel=3.5/stable
+                  - snap install microk8s --channel=1.29-strict/stable
+                  - snap install --classic charmcraft
+                  - snap refresh
+
+                runcmd:
+                - |
+                  # Clone source from the appropriate branch/commit
+                  sudo -u ubuntu git clone \
+                    -b ${{ github.head_ref || github.ref_name }} \
+                    https://github.com/canonical/intel-device-plugins-k8s-operator.git \
+                    ~ubuntu/intel-device-plugins-k8s-operator
```

Comment:
is that a typo `~ubuntu` ?

Reply:
No, that's using tilde expansion, see: https://stackoverflow.com/questions/998626/meaning-of-tilde-in-linux-bash-not-home-directory

Reply:
Great; i do not know it; thanks